pkg/cvo/availableupdates.go

```diff
@@ -176,29 +182,59 @@ func checkForUpdate(clusterID string, proxyURL *url.URL, upstream, channel strin
 	if len(upstream) == 0 {
 		return nil, fmt.Errorf("no upstream URL set for cluster version")
 	}
-	return cincinnati.NewClient(uuid, proxyURL).GetUpdates(upstream, channel, currentVersion)
+	return cincinnati.NewClient(uuid, proxyURL, tlsConfig).GetUpdates(upstream, channel, currentVersion)
 }
 
 // getHTTPSProxyURL returns a url.URL object for the configured
 // https proxy only. It can be nil if does not exist or there is an error.
-func (optr *Operator) getHTTPSProxyURL() (*url.URL, error) {
+func (optr *Operator) getHTTPSProxyURL() (*url.URL, configv1.ConfigMapNameReference, error) {
 	proxy, err := optr.proxyLister.Get("cluster")
 
 	if errors.IsNotFound(err) {
-		return nil, nil
+		return nil, configv1.ConfigMapNameReference{}, nil
 	}
 	if err != nil {
-		return nil, err
+		return nil, configv1.ConfigMapNameReference{}, err
 	}
 
 	if &proxy.Spec != nil {
 		if proxy.Spec.HTTPSProxy != "" {
 			proxyURL, err := url.Parse(proxy.Spec.HTTPSProxy)
 			if err != nil {
-				return nil, err
+				return nil, configv1.ConfigMapNameReference{}, err
 			}
-			return proxyURL, nil
+			return proxyURL, proxy.Spec.TrustedCA, nil
 		}
 	}
-	return nil, nil
+	return nil, configv1.ConfigMapNameReference{}, nil
+}
+
+func (optr *Operator) getTLSConfig(cmNameRef configv1.ConfigMapNameReference) (*tls.Config, error) {
+	cm, err := optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Get(cmNameRef.Name, metav1.GetOptions{})
```

Comment:
https://github.com/openshift/cluster-version-operator/pull/213/commits/c14773282d0cdd2c5bcce6d5a29c7e2e274337e5

added an informer/lister for configmaps for `openshif-config` namespaces.. could we do somethings like that do that we are calling the api and using the cache if possible..